### ISSUE TITLE
path to mantl-api credential should be discoverable

### ISSUE THREAD

**Issue by ryane** [CiscoCloud/mantl-universe#23]
The path to the mantl-api mesos credential file should not be hardcoded with `/etc/mesos/` in packages that require a credential file:
- [x] elasticsearch
- [x] chronos

**Comment by thothd** [comment]
@ryane I actually just bumped into the bug, I can't deploy the ElasticSearch framework (via the Mantl API), getting a java.io.FileNotFoundException: /etc/mesos/mantl-api.
what's the simplest way to overcome the issue?

**Comment by ryane** [comment]
ah, are you running mantl from master? the location of the credential file did change with https://github.com/CiscoCloud/microservices-infrastructure/pull/951. Also, the packaged version (0.5.2) of the Elasticsearch framework in this repo doesn't work with mesos 0.25 which is the version on Mantl master. I will work on building a new package that is compatible (#25).

In the meantime, there are a couple of changes you can try, if you want. You will need to update the mantl.json and the marathon.json for the elasticsearch package in the Consul KV store (something like: https://control-node/consul/#/your-dc/kv/mantl-install/repository/1/repo/packages/E/elasticsearch/0/)
1. mantl.json
   
   This bumps the version of the framework to 0.7.0.
   
   ``` json
   {
     "mesos": {
       "master": "zk://zookeeper.service.consul:2181/mesos"
     },
     "elasticsearch": {
       "executor": {
         "cluster-name": "mantl",
         "image": "mesos/elasticsearch-executor:0.7.0"
       },
       "scheduler": {
         "image": "mesos/elasticsearch-scheduler:0.7.0"
        }
     }
   }
   ```
2. marathon.json
   
   This fixes the credential path and removes some command line arguments (`zookeeperFrameworkTimeout` and `zookeeperFrameworkUrl`) that are no longer present in the 0.7.0 version.
   
   ``` json
   {
     "id": "{{elasticsearch.framework-name}}",
     "cpus": {{elasticsearch.scheduler.cpu}},
     "mem": {{elasticsearch.scheduler.ram}},
     "ports": [0],
     "instances": {{elasticsearch.scheduler.instances}},
     "healthChecks": [
       {
         "gracePeriodSeconds": 120,
         "intervalSeconds": 30,
         "maxConsecutiveFailures": 0,
         "path": "/",
         "portIndex": 0,
         "protocol": "HTTP",
         "timeoutSeconds": 5
       }
     ],
     "args": [
       "--zookeeperMesosUrl", "{{mesos.master}}",
       "--zookeeperMesosTimeout", "{{elasticsearch.zookeeper-mesos-timeout}}",
       "--webUiPort", "{{elasticsearch.port}}",
       "--frameworkFailoverTimeout", "{{elasticsearch.failover-timeout}}",
       "--frameworkName", "{{elasticsearch.framework-name}}",
   
       "--elasticsearchNodes", "{{elasticsearch.executor.instances}}",
       "--elasticsearchClusterName", "{{elasticsearch.executor.cluster-name}}",
       "--elasticsearchCpu", "{{elasticsearch.executor.cpu}}",
       "--elasticsearchDisk", "{{elasticsearch.executor.disk}}",
       "--elasticsearchRam", "{{elasticsearch.executor.ram}}",
       "--elasticsearchSettingsLocation", "{{elasticsearch.executor.settings-location}}",
   
       "--executorForcePullImage", "{{elasticsearch.force-pull-image}}",
       "--executorHealthDelay", "{{elasticsearch.executor.health-delay}}",
       "--executorImage", "{{elasticsearch.executor.image}}",
       "--executorName", "{{elasticsearch.executor.name}}",
       "--executorTimeout", "{{elasticsearch.executor.timeout}}"{{#mantl.mesos.authentication-enabled}},
       "--frameworkPrincipal", "{{mantl.mesos.principal}}",
       "--frameworkSecretPath", "/etc/mesos/{{mantl.mesos.principal}}"{{/mantl.mesos.authentication-enabled}}
     ],
     "env": {
       "JAVA_OPTS": "{{elasticsearch.scheduler.java-heap}}"
     },
     "container": {
       "type": "DOCKER",
       "docker": {
         "image": "{{elasticsearch.scheduler.image}}",
         "network": "HOST",
         "forcePullImage": {{elasticsearch.force-pull-image}}
       }{{#mantl.mesos.authentication-enabled}},
       "volumes": [
         {
           "containerPath": "/etc/mesos/{{mantl.mesos.principal}}",
           "hostPath": "/etc/sysconfig/{{mantl.mesos.principal}}",
           "mode": "RO"
         }
       ]{{/mantl.mesos.authentication-enabled}}
     },
     "ports": [{{elasticsearch.port}}],
     "requirePorts": true
   }
   ```

**Comment by thothd** [comment]
yeah, I'm running mantl from master.
I just started off from there since I wanted to latest versions of Mesos/Marathon + some other fixes.
Thanks for the workaround, I'll hack my way through it.

**Comment by ryane** [comment]
this was is resolved in the 0.6 version of the repo (default as of mantl-api 0.1.4)
